server/pulp/server/managers/content/orphan.py

```diff
@@ -256,13 +259,42 @@ def delete_orphan_content_units_by_type(type_id):
             for non_orphan_id in non_orphan:
                 unit_dict.pop(non_orphan_id)
 
-            # Remove the unit and any references on disk
+            # If there's a list of unit ids to delete, filter the orphan list
+            if content_unit_ids is not None:
+                for orphaned_unit_id in unit_dict.keys():
+                    if orphaned_unit_id not in content_unit_ids:
```

Comment:
Unless you only expect to have a handful at a time, this will perform better if ``content_unit_ids`` is a set.